app/ts/components/subcomponents/PreactBlocky.tsx

```diff
@@ -111,40 +109,39 @@ function generateIdenticon(options: BlockieProps, canvasRef: HTMLCanvasElement)
 	}
 
 	const opts = options || {}
-	const size = opts.size || 8
 	const scale = opts.scale || 4
-	const seed = opts.seed || Math.floor((Math.random()*Math.pow(10,16))).toString(16)
+	const seed = addressString(opts.address)
 
 	seedrand(seed)
 
 	const color = opts.color || createColor()
-	const bgcolor = opts.bgColor || createColor()
-	const spotcolor = opts.spotColor || createColor()
-	const imageData = createImageData(size)
+	const bgcolor = createColor()
+	const spotcolor = createColor()
+	const imageData = createImageData(8)
 	const canvas = setCanvas(canvasRef, imageData, color, scale, bgcolor, spotcolor)
 
 	return canvas
 }
 
 export default function Blockie(props: BlockieProps) {
 	const scale = props.scale || 4
-	const dimension = (props.size || 8) * scale
-	const [seed, setSeed] = useState<string | undefined>(props.seed)
-	const [dataURL, setDataURL] = useState<string | undefined>(dataURLCache.get(`${ props.seed }!${ dimension }`))
+	const dimension = 8 * scale
+	const [address, setAddress] = useState<bigint | undefined>(props.address)
```

Comment:
Why do you store the address (which is already coming from props) in local state?  This will make it so a change of address on the `<Blockie .../>` will not correctly propagate to update the blockie.

Reply:
this is only called when the element is created I think?

Reply:
This example component has its address changed every second(ish), but the Blockie won't update with the code as you have it now I don't think.
```tsx
function MyComponent() {
	const address = useSignal(0n)
	useEffect(() => {
		const id = setTimeout(() => address.value = Date.now % 2 ? 1n : 0n = , 1000)
		return () => clearTimeout(id)
	})
	return <Blockie address={address.value}/>
}
```